priam/src/main/java/com/netflix/priam/backupv2/MetaFileWriter.java

```diff
@@ -0,0 +1,155 @@
+     *
+     * @param columnfamilyResult a POJO encapsulating the column family result
+     * @throws IOException
+     */
+    public void addColumnfamilyResult(ColumnfamilyResult columnfamilyResult) throws IOException {
+        if (jsonWriter != null && columnfamilyResult != null)
+            jsonWriter.jsonValue(columnfamilyResult.toString());
+    }
+
+    /**
+     * Finish the generation of meta.json file and save it on local media.
+     *
+     * @return {@link Path} to the local meta.json produced.
+     * @throws IOException
+     */
+    public Path endMetaFileGeneration() throws IOException {
+        if (jsonWriter == null)
+            return null;
+
+        jsonWriter.endArray();
+        jsonWriter.endObject();
+        jsonWriter.close();
+
+        Path tempMetaFilePath = Paths.get(metaFileDirectory.toString(), metaFilePath.toFile().getName() + ".tmp");
+
+        //Rename the tmp file.
+        tempMetaFilePath.toFile().renameTo(metaFilePath.toFile());
+        logger.info("Finished writing to meta file: {}", metaFilePath);
+
+        return metaFilePath;
+    }
+
+    /**
+     * Upload the meta file generated to backup file system.
+     *
+     * @param metafile        {@link Path} to the local meta file that needs to be backed up.
+     * @param deleteOnSuccess delete the meta file from local file system if backup is successful. Useful for testing purposes
+     * @throws Exception when unable to upload the meta file.
+     */
+    public void uploadMetaFile(Path metafile, boolean deleteOnSuccess) throws Exception {
+        AbstractBackupPath abstractBackupPath = pathFactory.get();
+        abstractBackupPath.parseLocal(metafile.toFile(), AbstractBackupPath.BackupFileType.META_V2);
+        new RetryableCallable<Void>() {
+            @Override
+            public Void retriableCall() throws Exception {
+                backupFileSystem.upload(abstractBackupPath, abstractBackupPath.localReader());
+                return null;
+            }
+        }.call();
+        abstractBackupPath.setCompressedFileSize(backupFileSystem.getBytesUploaded());
+
+        if (deleteOnSuccess)
+            FileUtils.deleteQuietly(metafile.toFile());
+    }
+
+    /**
+     * Delete the old meta files, if any present in the metaFileDirectory
+     */
+    public void cleanupOldMetaFiles(){
+        IOFileFilter fileNameFilter = FileFilterUtils.and(FileFilterUtils.prefixFileFilter(MetaFileInfo.META_FILE_PREFIX),
+                FileFilterUtils.or(FileFilterUtils.suffixFileFilter(MetaFileInfo.META_FILE_SUFFIX),
+                        FileFilterUtils.suffixFileFilter(MetaFileInfo.META_FILE_SUFFIX + ".tmp")));
+        Collection<File> files = FileUtils.listFiles(metaFileDirectory.toFile(), fileNameFilter, null);
+        files.stream().filter(file -> file.isFile()).forEach(file -> file.delete());
```

Comment:
Might want to add a `warn/debug` here mentioning deletion of metafiles, will be helpful in debugging.